src/vsc_equiv.erl

```diff
@@ -57,13 +60,65 @@ check_equiv(OrigHash, RefacHash, OrigName, RefacName) ->
     % Generate random data (assume we know the type of data for now) and check
     % if the outputs are equivalent
     % RES = proper:quickcheck(?FORALL(I, integer(), prop_same_output(OrigNode, RefacNode, test, f_old, f_new, [I]))), % TODO Give back the minimal input that falsifies the assertion
-    RES = proper:quickcheck(?FORALL(L, list(integer()), prop_same_output(OrigNode, RefacNode, test, f_old, f_new, [L]))), % TODO Give back the minimal input that falsifies the assertion
+    % RES = proper:quickcheck(?FORALL(L, list(integer()), prop_same_output(OrigNode, RefacNode, test, OrigName, RefacName, [L]))), % TODO Give back the minimal input that falsifies the assertion
+
+    file:set_cwd(".."),
+    cleanup(),
+    stop_nodes(OrigNode, RefacNode),
+    application:stop(wrangler). % TODO
+    %RES.
+
+check_equiv(OrigHash, RefacHash) ->
+    application:start(wrangler), % TODO
+    {_, ProjFolder} = file:get_cwd(),
+    recreate_project(ProjFolder),
+    file:set_cwd("tmp"), % TODO Change this to something like /tmp later
+
+    % Module
+    % Function names (old and new)
+    % Arity
+
+    % Caller functions
+    
+    % -spec info
+
+    {ChangedFile, {OrigFun, RefacFun}, Arity} = general_refac:diff_renaming(OrigHash, RefacHash),
+    Callers = general_refac:find_callers({RefacFun, Arity}),
+
+    CallerFiles = lists:map(fun({FileName, _, _}) -> FileName end, Callers),
+
+    Modules = lists:uniq([ChangedFile|CallerFiles]),
+
+    % Checkout and compile the necessary modules into two separate folders
+    % This is needed because QuickCheck has to evaluate to old and the new
+    % function repeatedly side-by-side
+    checkout(OrigHash),
+    comp(Modules, "orig"),
+
+    checkout(RefacHash),
+    comp(Modules, "refac"),
+
+    {OrigNode, RefacNode} = start_nodes(),
+
+    % Contains all the functions that call the renamed one {Module, Function, PropEr Type}
+    Funs = lists:map(fun({FileName,F,A}) -> {get_module(FileName),erlang:list_to_atom(F),get_type(hd(general_refac:get_args(FileName,F,A)))} end, Callers),
+
+    lists:map(fun({M,F,Type}) -> proper:quickcheck(?FORALL(X, Type, prop_same_output(OrigNode, RefacNode, M, F, [X]))) end, Funs),
 
     file:set_cwd(".."),
     cleanup(),
     stop_nodes(OrigNode, RefacNode),
-    application:stop(wrangler), % TODO
-    RES.
+    application:stop(wrangler). % TODO
+    % RES.
+
+get_type(T) ->
+    if
+        T =:= "list(integer())" -> proper_types:list(integer());
+        T =:= "integer()" -> proper_types:integer()
+    end.
```

Comment:
This function is only for "translating" the type information I get from the `-spec` declaration to the appropriate PropEr type, because I don't know this at runtime.

Reply:
PropEr has a function in its typeserver module that does exactly this: `demo_translate_type/2`.
Although they suggest that it should only be used for demonstrational purposes.
The intended way to use PropEr is to just write the tests alongside the codebase, so you know the type in advance, but I have to to this dynamically.

Reply:
Test if `list(list(int()))` works.

Reply:
https://github.com/proper-testing/proper/blob/master/src/proper_typeserver.erl#L2091

Reply:
I was able to reuse the `translate_type` function from the typeserver to automatically convert types.
More complex types like `list(list(integer()))` also work fine.